include/benchmark/benchmark_api.h

```diff
@@ -609,11 +756,15 @@ class Fixture: public internal::Benchmark {
 
     virtual void Run(State& st) {
       this->SetUp(st);
+      this->InitState(st);
       this->BenchmarkCase(st);
+      this->TerminateState(st);
       this->TearDown(st);
     }
 
     virtual void SetUp(const State&) {}
+    virtual void InitState(State &) {}
```

Comment:
These were needed to be able to change the state before and after the benchmark, AND outside of the benchmark. It wouldn't be possible to start or stop the counters in SetUp() and TearDown() because they receive a const reference to State, which is where the counters are. The solutions could be:

- Remove const from the State argument to the SetUp() and TearDown() of Fixture. I chose not to do this because it was more disruptive to how the public API was. But if this was my own project, I would have gone for this.
- Make the counters interface mutable in the State class, ie add const to the methods adding counters. I chose not to do this because it would change the const semantics of a significant part of the class.
- Finally, adding these two methods. This has the advantage of being merely additive to the existing API and does not have the constness problems above.

If you prefer a different solution let me know!